spec/aptible/cli/helpers/s3_log_helpers_spec.rb

```diff
@@ -0,0 +1,320 @@
+require 'spec_helper'
+
+describe Aptible::CLI::Helpers::S3LogHelpers do
+  subject { Class.new.send(:include, described_class).new }
+  let(:v2_pfx) { 'mystack/shareable/v2/fakesha' }
+  let(:v3_pfx) { 'mystack/shareable/v3/fakesha' }
+  let(:v2app) do
+    "#{v2_pfx}/apps-321/fakebread-json.log.2022-06-29T18:30:01.bck.gz"
+  end
+  let(:v2app_rotated) do
+    "#{v2_pfx}/apps-321/fakebread-json.1.log.2022-06-29T18:30:01.bck.gz"
+  end
+  let(:v3app) do
+    "#{v3_pfx}/apps-321/service-123/deadbeef-json.log." \
+    '2022-08-24T21:12:33.2022-08-24T21:14:38.archived.gz'
+  end
+  let(:v3db) do
+    "#{v3_pfx}/databases-321/fakebread-json.log." \
+    '2022-08-24T21:12:33.2022-08-24T21:14:38.archived.gz'
+  end
+  let(:v3db_rotated) do
+    "#{v3_pfx}/databases-321/fakebread-json.log.1." \
+    '2022-08-24T21:12:33.2022-08-24T21:14:38.archived.gz'
+  end
+
+  describe '#ensure_aws_creds' do
+    it 'Raises if no keys are provided via ENV' do
+      expect { subject.ensure_aws_creds }
+        .to raise_error(Thor::Error, /Missing environment variable/)
+    end
+
+    it 'Accepts AWS keypair from the ENV' do
+      ENV['AWS_ACCESS_KEY_ID'] = 'foo'
+      ENV['AWS_SECRET_ACCESS_KEY'] = 'bar'
+      expect { subject.ensure_aws_creds }.to_not raise_error
+    end
+  end
+
+  describe '#info_from_path' do
+    context 'time zones are in UTC' do
+      it 'processes v2 upload time in UTC' do
+        result = subject.info_from_path(v2app)
+        expect(result[:uploaded_at].zone).to eq('UTC')
+      end
+
+      it 'processes v3 log times in UTC' do
+        result = subject.info_from_path(v3app)
+        expect(result[:start_time].zone).to eq('UTC')
+        expect(result[:end_time].zone).to eq('UTC')
+      end
+    end
+
+    it 'can read app data from v2 paths' do
+      result = subject.info_from_path(v2app)
+      expect(result[:schema]).to eq('v2')
+      expect(result[:shasum]).to eq('fakesha')
+      expect(result[:type]).to eq('apps')
+      expect(result[:id]).to eq(321)
+      expect(result[:service_id]).to be(nil)
+      expect(result[:container_id]).to eq('fakebread')
+      expect(result[:uploaded_at]).to eq('2022-06-29T18:30:01')
+      expect(result[:container_id]).to eq('fakebread')
+      expect(result[:start_time]).to be(nil)
+      expect(result[:end_time]).to be(nil)
+    end
+
+    it 'can read app data from v3 paths' do
+      result = subject.info_from_path(v3app)
+      expect(result[:schema]).to eq('v3')
+      expect(result[:shasum]).to eq('fakesha')
+      expect(result[:type]).to eq('apps')
+      expect(result[:id]).to eq(321)
+      expect(result[:service_id]).to eq(123)
+      expect(result[:container_id]).to eq('deadbeef')
+      expect(result[:uploaded_at]).to be(nil)
+      expect(result[:start_time]).to eq('2022-08-24T21:12:33')
+      expect(result[:end_time]).to eq('2022-08-24T21:14:38')
+    end
+
+    it 'can read db data from v3 paths' do
```

Comment:
Now that it catches `unknown` times it may be a good idea to test that path as well.

Reply:
:+1: